site_scons/prereq_tools/base.py

```diff
@@ -809,10 +819,10 @@ def _setup_compiler(self, warning_level):
 
         compiler = self.__env.get('COMPILER').lower()
         if compiler == 'icc':
-            self._setup_intelc()
+            compiler_map['icc'] = self._setup_intelc()
 
         if warning_level == 'error':
-            if compiler == 'icc':
+            if compiler == 'icc' and not self.has_icx:
```

Comment:
what does icx support, -Werror or -Werror-all or both?  Changing this from -Werror-all to -Werror might be why we don't need to the -diag-disable lines above?

Reply:
-Werror-all is not recognized by icx